nym-vpn-android/nym-vpn-client/src/main/java/net/nymtech/vpn/util/AllowedIpsCalculator.kt

```diff
@@ -0,0 +1,195 @@
+package net.nymtech.vpn.util
+
+import kotlin.math.floor
+import kotlin.math.ln
+import kotlin.math.pow
+
+// TODO this needs ipv6 support added
+object AllowedIpsCalculator {
```

Comment:
Would have thought something like this would already be implemented in libraries, or at least the building blocks to implement it without the detailed manual manipulations

Reply:
Same! I was surprised to find there was nothing. I think I will eventually refactor it and publish it as a library. There was only a couple of repos that generate this allowedIps list (which I based this off of), and most are in python.